include/gc/Dialect/OneDNNGraph/OneDNNGraphTypes.td

```diff
@@ -17,14 +17,26 @@ include "OneDNNGraphDialect.td"
 // OneDNNGraph type definitions
 //===----------------------------------------------------------------------===//
 
+//===----------------------------------------------------------------------===//
+// Floating-point types.
+//===----------------------------------------------------------------------===//
+def OneDNNGraph_Float : AnyTypeOf<[F32,
+			                       F16,
+			                       BF16]>;
+
+//===----------------------------------------------------------------------===//
+// Integer types.
+//===----------------------------------------------------------------------===//
+
+def OneDNNGraph_Int : AnyTypeOf<[SI<8>,
+                                 UI<8>]>;
```

Comment:
What is the rationale for having separate types? Also, why do we only have fp16 and int8?

Reply:
Easier to construct Quantize/Dequantize ops later. https://oneapi-src.github.io/oneDNN/dev_guide_op_quantize.html#supported-data-types